src/gui/guiutility.cpp

```diff
@@ -543,6 +544,45 @@ void GuiUtility::invalidateLayout(QLayout *layout) {
     layout->activate();
 }
 
+bool GuiUtility::isSyncFolderValid(const QString &dirPath, const std::map<int, SyncInfoClient> &syncInfoMap, QWidget *parent) {
```

Comment:
```suggestion
bool GuiUtility::warnOnInvalidSyncFolder(const QString &dirPath, const std::map<int, SyncInfoClient> &syncInfoMap, QWidget *parent) {
```

Reply:
Alternatively, we may design a function that returns a `QString` containing a warning message, if any, empty otherwise. Then the caller instantiate a `CustomMessageBox` if the warning message is not empty.

Reply:
Done